autodispose/src/main/java/com/uber/autodispose/ParallelFlowableScoper.java

```diff
@@ -0,0 +1,50 @@
+package com.uber.autodispose;
+
+import org.reactivestreams.Subscriber;
+
+import io.reactivex.Maybe;
+import io.reactivex.functions.Function;
+import io.reactivex.parallel.ParallelFlowable;
+
+class ParallelFlowableScoper<T> extends Scoper
+    implements Function<ParallelFlowable<? extends T>, ParallelFlowableSubscribeProxy<T>> {
+
+  ParallelFlowableScoper(Maybe<?> lifecycle) {
+    super(lifecycle);
+  }
+
+  @Override public ParallelFlowableSubscribeProxy<T> apply(final ParallelFlowable<? extends T> source)
+      throws Exception {
+    return new ParallelFlowableSubscribeProxy<T>() {
+      @Override
+      public void subscribe(Subscriber<? super T>[] subscribers) {
+        new AutoDisposeParallelFlowable<>(source, scope()).subscribe(subscribers);
+      }
+    };
+  }
+
+  static final class AutoDisposeParallelFlowable<T> extends ParallelFlowable<T> {
+
+    private final ParallelFlowable<T> source;
+    private final Maybe<?> scope;
+
+    AutoDisposeParallelFlowable(ParallelFlowable<T> source, Maybe<?> scope) {
+      this.source = source;
+      this.scope = scope;
+    }
+
+    @Override public void subscribe(Subscriber<? super T>[] subscribers) {
+      Subscriber<? super T>[] newSubscribers = new Subscriber[subscribers.length];
+      for (int i = 0; i < subscribers.length; i++) {
```

Comment:
Interesting... I'm not super familiar with `ParallelFlowable`, but are there any thread safety concerns here? Is this how other `subscribe()` calls are handled in its operators?

Reply:
Yep, that's what is happening in the `subscribe()` method of the operators. They iterate over the passed in array of subscribers, create a new array subscribers and pass it to the source flowable. 

One very important thing which I forgot to do was checking if the number of subscribers is equal to the parallelism of the flowable (`subscribe()'s` contract says that they should be equal.) I'll push another commit tonight.